### PR TITLE
Bug 1219944: document a convention

### PR DESCRIPTION
**Pull request by djmitche**

### CODE REVIEW

File: devel/conventions/index.md [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+## Scopes and Roles
+
+### When To Require A Role
+
+When verifying the scopes for an API call or some other operation, the default behavior should be to check for appropriately-parameterized scopes.
+These answer the question, "Can the caller do X".
+
+However, when the operation modifies an object which will later perform actions using a role, then it is appropriat to require that the caller posess that role.
+For example, hooks trigger tasks using role `hook-id<hookGroupId>/<hookId>`, so the API calls to create and modify the hook require `assume:hook-id:<hookGroupId>/<hookId>` directly.
+It is not enough simply to possess the role's extended scopes -- the caller must possess the assumed role itself.
```
Comment (jonasfj) [inline devel/conventions/index.md]: expanded scopes

File: devel/conventions/index.md [hunk 2]
```diff
@@ -0,0 +1,20 @@
+---
+layout:   default
+class:    markdown
+---
+
+Conventions
+===========
+
+TaskCluster is a very open platform, and can easily descend into anarchy if we are not careful to enforce some standard approaches and conventions.
```
Comment (selenamarie) [inline devel/conventions/index.md]: How about: "TaskCluster is a very open platform! To help us with our day to day work, we are setting up a few standard approaches and conventions to help make administration and coordination between the humans using the platform easier."

Comment (jonasfj) [inline devel/conventions/index.md]: "Loosely typed" or " dynamic platform" also works...

Note, these conventions are just as much for us, as they are for customers... In fact they are mainly relevant to us and maybe power users..

**Comment by jonasfj** [comment]
There is a lot more to add here so far I like it...

File: devel/conventions/index.md (hunk 2 above)
Comment (djmitche) [inline devel/conventions/index.md]: Good advice - thanks.

**Comment by djmitche** [comment]
Yeah, this is mostly getting the section started.  I'm sure there will be more PRs :)

**Comment by selenamarie** [comment]
:tada: +1

**Pull request merged**